DOKS-automatic-node-repair/README.md

```diff
@@ -1 +1,374 @@
+
+   ```
+   NAME            DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                 AGE
+   cilium          2         2         2       2            2           <none>                        22h
+   csi-do-node     2         2         2       2            2           <none>                        2d21h
+   do-node-agent   2         2         2       2            2           beta.kubernetes.io/os=linux   2d21h
+   doks-debug      2         2         2       2            2           <none>                        112m
+   kube-proxy      2         2         2       2            2           <none>                        44h
+   ```
+
+3. Verify the `doks-debug` Pods:
+
+   ```bash
+   kubectl get pods -l name=doks-debug -n kube-system
+   ```
+
+   The output looks similar to the following:
+
+   ```
+   NAME               READY   STATUS    RESTARTS   AGE
+   doks-debug-m6xlj   1/1     Running   0          105m
+   doks-debug-qgw2m   1/1     Running   0          115m
+   ```
+
+If everything looks good then we're all set and we can proceed with the next steps.
+
+### Killing the Kubelet
+
+For this to happen we have to `exec` into one of the `doks-debug` pods and get access to worker node system services. Once there we can simply stop the `kubelet` service and watch how the node goes away from the `kubectl get nodes` command output.
+
+The experiment steps:
+
+1. Open a new terminal window and issue a `watch` for the worker nodes:
+
+    ```bash
+    watch "kubectl get nodes"
+    ```
+
+    **Note:**
+
+    If the `watch` command is not available it can be installed very easy - for example on MacOS, `Homebrew` can be used:
+
+    ```bash
+    brew install watch
+    ```
+2. Pick a `doks-debug` Pod and `exec` into it:
+
+    ```bash
+    DOKS_DEBUG_POD_NAME=$(kubectl get pods -l name=doks-debug -ojsonpath='{.items[0].metadata.name}' -n kube-system)
+    kubectl exec -it "$DOKS_DEBUG_POD_NAME" -n kube-system -- bash
+    ```
+    If everything is ok we should be presented with a prompt that looks similar to:
+
+    ```
+    root@basicnp-8hx1a:~#
+    ```
+3. Let's `chroot` and inspect the `kubectl` system service:
+
+    ```bash
+    chroot /host /bin/bash
+    systemctl status kubelet
+    ```
+
+    The output should look similar to the following:
+
+    ```
+    ● kubelet.service - Kubernetes Kubelet Server
+   Loaded: loaded (/etc/systemd/system/kubelet.service; enabled; vendor preset: enabled)
+   Active: active (running) since Fri 2021-08-06 08:48:42 UTC; 2h 18min ago
+    Docs: https://kubernetes.io/docs/concepts/overview/components/#kubelet
+    Main PID: 1053 (kubelet)
+        Tasks: 17 (limit: 4701)
+    Memory: 69.3M
+    CGroup: /system.slice/kubelet.service
+            └─1053 /usr/bin/kubelet --config=/etc/kubernetes/kubelet.conf --logtostderr=true --image-pull-progress-deadline=5m
+    ...
+    ```
+4. Let's stop the `kubelet`:
+
+    ```bash
+    systemctl stop kubelet
+    ```
```

Comment:
A more concise alternative to the interactive `kubectl exec` example would be a one-liner like

```shell
kubectl exec -it "$DOKS_DEBUG_POD_NAME" -n kube-system -- chroot /host systemctl stop kubelet
```

(Untested, but done in the past so it should work in general).

Reply:
Thanks for the suggestion. I added it in a dedicated `Tips` section.